server/plugins/api.js

```diff
@@ -1,19 +1,18 @@
-const fetch = require('node-fetch')
+const request = require('request-promise-native')
+const config = require('../config')
 
 module.exports = {
   plugin: {
     name: 'api',
     register: (server, options) => {
       server.method({
-        name: 'api.getJson',
-        method: async (url) => new Promise(async (resolve, reject) => {
-          try {
-            const response = await fetch(url)
-            resolve(await response.json())
-          } catch (error) {
-            reject(error)
-          }
-        }),
+        name: 'api.get',
+        method: async (action) => {
```

Comment:
you could remove async from the method call and return the  request without the await, which means all the api plugin does is wrap the function `action => request({ json: true, url: config.apibase + action)`

That's a lot of boilerplate to create a function that is accessed by the indirection of `request.server.methods.api.get`.
I'd be tempted not to use an API plugin at all and create a simple API Service library that exposed a `getViewCount` function, and any other methods required.
The advantage of this over a HAPI plugin it that it is explicitly required at the top of the file, making it easier to see what is used by the code.

Reply:
I agree with your points and would like to be corrected if I've misread something (because this feels a bit silly to me) but my understanding from the Hapi documentation is that plugins are the recommended approach, rather than importing libraries directly throughout an app.

Reply:
Sure, it's not entirely clear what best practices are, perhaps we need a few pointers from Paul.

Looking at an existing project https://github.com/DEFRA/water-abstraction-ui/tree/develop/src/external/lib/hapi-plugins the plugins are very small focussed middleware for validation or routing, rather than being areas of the application.

They seem to have organised the application into modules that expose routes, but share a common logic in the 'lib' folder. If we took this approach then the getViewCount would either be in the module controller or lib rather than a plugin.

I think we need a development meeting to decide on project structure and to agree what belongs in a HAPI plugin.

Reply:
Definitely. Agreed we should get Paul involved in that meeting too.